src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx

```diff
@@ -73,13 +76,37 @@ type WeatherWidgetProps = { name: string, editIcao: string, icao: string};
 const WeatherWidget = (props: WeatherWidgetProps) => {
     const [metar, setMetar] = useState<MetarParserType>(MetarParserTypeProp);
     const [showMetar, setShowMetar] = usePersistentProperty(`CONFIG_SHOW_METAR_${props.name}`, 'DISABLED');
-    let [metarSource] = usePersistentProperty('CONFIG_METAR_SRC', 'MSFS');
+    const [coloredMetar, setColoredMetar] = useState('');
+    const [baroType] = usePersistentProperty('CONFIG_INIT_BARO_UNIT', 'HPA');
+    const getBaroTypeForAirport = (icao: string) => (['K', 'C', 'M', 'P', 'RJ', 'RO', 'TI', 'TJ']
+        .some((r) => icao.toUpperCase().startsWith(r)) ? 'IN HG' : 'HPA');
+    const [metarSource] = usePersistentProperty('CONFIG_METAR_SRC', 'MSFS');
+    const source = metarSource === 'MSFS' ? 'MS' : metarSource;
 
-    if (metarSource === 'MSFS') {
-        metarSource = 'MS';
-    }
+    const BaroValue = () => {
+        const displayedBaroType = baroType === 'AUTO' ? getBaroTypeForAirport(metar.icao) : baroType;
+        if (displayedBaroType === 'IN HG') {
+            return (
+                <>
+                    {metar.barometer.hg.toFixed(2)}
+                    {' '}
+                    inHg
+                </>
+            );
+        }
+        return (
+            <>
+                {metar.barometer.mb.toFixed(0)}
+                {' '}
+                mb
+            </>
+        );
+    };
 
-    const source = metarSource;
+    const MetarText = () => (
+        // eslint-disable-next-line react/no-danger
+        <span dangerouslySetInnerHTML={{ __html: coloredMetar }} />
```

Comment:
This is called "dangerous" for a reason. ***Especially*** since we are pulling this from online sources, therefore making it ***very unsafe*** to do this, as that is untrusted data.

Since we are already building HTML strings, why not use JSX instead ?

Reply:
Thanks, good feedback.
I refactored this so there is no further need for  dangerouslySetInnerHTML.